newscollector.js

```diff
@@ -188,39 +188,74 @@ app.get('/articles' , function(request,response){
         }).catch(function(err){
             console.log(err);
         });
-     
+
 
     }
 });
 
 app.get('/redir' , function(request,response){
-    response.redirect('/'+request.query['choice'])
+
+  //Fix the Open Redirect Vulnerability by sanitizing the request.query['choice'] variable.
+  //Use whitelist approach to define which paths are available for redirection to.
+
+  var redirectWhitelist = ["Sources", "all_articles"];//The only paths the application can redirect to.
```

Comment:
Variable names should use *[snake_case](https://en.wikipedia.org/wiki/Snake_case)* to match the rest of the code.